src/dp.rs

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Differential privacy (DP) primitives.
+use std::fmt::Debug;
+
+/// Positive rational number to represent DP and noise distribution parameters in protocol messages
+/// and manipulate them without rounding errors.
+#[allow(dead_code)]
+#[derive(Clone, Debug)]
+pub struct Rational {
+    numerator: u32,
+    denominator: u32,
```

Comment:
If these aren't `pub` then there's no way to construct a `Rational` from outside the crate. Additionally, if there are any restrictions on what a valid `Rational` is (for instance, is it OK for `numerator > denominator`?) then we should provide a constructor that enforces that.

Reply:
Indeed. I actually had a constructor but we removed it since it wasn't used. We don't have any constraint on the fields, so I'll just stick to public fields.